app/models/notification.rb

```diff
@@ -9,20 +9,41 @@ class Notification < ApplicationRecord
 
   include Token
 
-  def good_faith?
+  def should_display?
     case notifiable
     when Message
-      good_faith_message?
+      should_display_message?
     when Comment
-      good_faith_comment?
+      should_display_comment?
     end
   end
 
-  def good_faith_message?
+  def should_display_message?
     true
   end
 
-  def good_faith_comment?
+  def should_display_comment?
+    return false unless user_wants_notification?
+    return false unless is_high_quality?
+
+    true
+  end
+
+  private
+
+  def user_wants_notification?
+    comment = notifiable
+
+    # Check if this is a mention notification
+    if comment.plaintext_comment.match?(/\B[@~]#{Regexp.escape(user.username)}\b/)
```

Comment:
I just wrote this exact same regexp in `extras/markdowner.rb`, let me lift it up to a constant you can reuse.

Reply:
20d85ebf